activesupport/lib/active_support/i18n_railtie.rb

```diff
@@ -62,8 +62,8 @@ def self.initialize_i18n(app)
 
       if app.config.reloading_enabled?
         directories = watched_dirs_with_extensions(reloadable_paths)
-        reloader = app.config.file_watcher.new(I18n.load_path.dup, directories) do
-          I18n.load_path.keep_if { |p| File.exist?(p) }
+        reloader = app.config.file_watcher.new(I18n.load_path.select { |path| path.start_with?(Rails.root.to_s) }, directories) do
```

Comment:
Looking good. Can we extract `I18n.load_path.select { |path| path.start_with?(Rails.root.to_s) }` to a private method, and can you please add a changelog entry? Thanks!